### PR TITLE
fix(scheduler): cast port to an int from environment

### LINKED ISSUE #856: Apps with healthchecks from 2.0 are broken in 2.1
@bacongobbler 
I have so far produced this twice with 2 separate applications.  The only fix so far for me was to destroy the application.

Deployment
```
Creating build... ...Error: 
400 Bad Request
detail: public-v49-cmd (app::deploy): ('failed to create ReplicationController "public-v49-cmd" in Namespace "public": 422 status code 422\n{\'metadata\': {}, \'code\': 422, \'reason\': \'Invalid\', \'kind\': \'Status\', \'status\': \'Failure\', \'message\': \'ReplicationController "public-v49-cmd" is invalid: spec.template.spec.containers[0].livenessProbe.httpGet.port: Invalid value: "80": must be an IANA_SVC_NAME (at most 15 characters, matching regex [a-z0-9]([a-z0-9-]*[a-z0-9])*, it must contain at least one letter [a-z], and hyphens cannot be adjacent to other hyphens): e.g. "http"\', \'details\': {\'name\': \'public-v49-cmd\', \'causes\': [{\'message\': \'Invalid value: "80": must be an IANA_SVC_NAME (at most 15 characters, matching regex [a-z0-9]([a-z0-9-]*[a-z0-9])*, it must contain at least one letter [a-z], and hyphens cannot be adjacent to other hyphens): e.g. "http"\', \'reason\': \'FieldValueInvalid\', \'field\': \'spec.template.spec.containers[0].livenessProbe.httpGet.port\'}], \'kind\': \'ReplicationController\'}, \'apiVersion\': \'v1\'}', 'public-v49-cmd', 'public')
```

List
```
deis healthchecks:list -a public                                                                                    
=== public Healthchecks

--- Liveness
Initial Delay (seconds): 10
Timeout (seconds): 50
Period (seconds): 10
Success Threshold: 1
Failure Threshold: 3
Exec Probe: N/A
HTTP GET Probe: Path="/" Port=0 HTTPHeaders=[]
TCP Socket Probe: N/A

--- Readiness
No readiness probe configured.
```

Unset and set fail
```
deis healthchecks:unset liveness -a public                                                                            
Removing healthchecks... Error: This action could not be completed due to a conflict.
```

### PR DESCRIPTION
**Pull request by bacongobbler**
Kubernetes requires the port number to be an integer or else it assumes that it is a named port.

fixes #856

### CODE REVIEW

File: rootfs/api/tests/test_release.py [hunk 1]
```diff
@@ -373,3 +373,42 @@ def test_release_no_change(self, mock_requests):
...
+        url = '/v2/apps/{app_id}/config'.format(**locals())
+        body = {'values': json.dumps({'PORT': '8080'})}
+        response = self.client.post(url, body)
+        self.assertEqual(response.status_code, 201, response.data)
+        release = app.release_set.latest()
+
+        # check that the port number returned is an int, not a string
+        self.assertEqual(release.get_port(routable=True), 8080)
+
+        # TODO(bacongobbler): test dockerfile ports
```
Comment (helgi) [inline rootfs/api/tests/test_release.py]: Are you going to finish this before merging or is this a punt?

Comment (bacongobbler) [inline rootfs/api/tests/test_release.py]: I'm punting this. I'll file a new ticket after this is merged, but I was just adding coverage for the issue as needed and I couldn't figure out how to test `get_port` with a dockerfile app cleanly within a reasonable timeframe. Gotta get on other sprint-related work.

**Comment by bacongobbler** [comment]
error logs:

```
18s	5s	2	{kubelet gke-venoge-default-pool-17b5d202-y1dq}	spec.containers{deis-controller}	Normal	Pulling		pulling image "quay.io/deis/controller:git-d7d1c9d"
17s	4s	2	{kubelet gke-venoge-default-pool-17b5d202-y1dq}	spec.containers{deis-controller}	Warning	Failed		Failed to pull image "quay.io/deis/controller:git-d7d1c9d": image pull failed for quay.io/deis/controller:git-d7d1c9d, this may be because there are no credentials on this request.  details: (Tag git-d7d1c9d not found in repository quay.io/deis/controller)
17s	4s	2	{kubelet gke-venoge-default-pool-17b5d202-y1dq}						Warning	FailedSync	Error syncing pod, skipping: failed to "StartContainer" for "deis-controller" with ErrImagePull: "image pull failed for quay.io/deis/controller:git-d7d1c9d, this may be because there are no credentials on this request.  details: (Tag git-d7d1c9d not found in repository quay.io/deis/controller)"
```

**Comment by bacongobbler** [comment]
And the error from the [controller build](https://ci.deis.io/job/controller-pr/911/console):

```
14:05:48 docker push deisci/controller:git-d7d1c9d
14:05:48 The push refers to a repository [docker.io/deisci/controller]
14:05:49 8c1fc4ab0488: Preparing
14:05:49 6b772acdb3a3: Preparing
14:05:49 056edaa0a2eb: Preparing
14:05:49 9eb130d61eec: Preparing
14:05:49 f7e7f3d0c903: Preparing
14:05:49 5f70bf18a086: Preparing
14:05:49 f34e865127cf: Preparing
14:05:49 5f70bf18a086: Waiting
14:05:49 f34e865127cf: Waiting
14:05:50 6b772acdb3a3: Layer already exists
14:05:50 6b772acdb3a3: Layer already exists
14:05:50 8c1fc4ab0488: Layer already exists
14:05:50 8c1fc4ab0488: Layer already exists
14:05:50 f7e7f3d0c903: Layer already exists
14:05:50 f7e7f3d0c903: Layer already exists
14:05:50 056edaa0a2eb: Layer already exists
14:05:50 056edaa0a2eb: Layer already exists
14:05:50 9eb130d61eec: Layer already exists
14:05:50 9eb130d61eec: Layer already exists
14:05:50 f34e865127cf: Layer already exists
14:05:50 f34e865127cf: Layer already exists
14:05:50 5f70bf18a086: Layer already exists
14:05:50 5f70bf18a086: Layer already exists
14:05:51 git-d7d1c9d: digest: sha256:d161ee40d57dd3ab8c95855ccc514ea9aa078f47caf8904b069c338eedf265c2 size: 1760
14:05:52 WARNING: login credentials saved in /home/jenkins/.docker/config.json
14:05:52 Login Succeeded
14:06:23 Unable to connect to the server: dial tcp 104.197.71.127:443: i/o timeout
```

CC'ing @vdice on this for visibility. Will file an issue :)

**Pull request merged**